Exception is thrown when application is closed on android devices

![ezgif-3871036910](https://cloud.githubusercontent.com/assets/2364994/18809486/359260f6-8285-11e6-9430-c56578dfacd3.gif)

```
09-24 11:19:16.963    99    99 D Genyd   : Received Set Clipboard
09-24 11:19:16.963    99    99 D Genymotion: Received Set Clipboard
09-24 11:19:18.394   676  1164 I ActivityManager: START u0 {act=com.android.systemui.recents.SHOW_RECENTS flg=0x10804000 cmp=com.android.systemui/.recents.RecentsActivity} from uid 10016 on display 0
09-24 11:19:18.431   676  1164 D         : HostConnection::get() New Host Connection established 0xd9491f40, tid 1164
09-24 11:19:18.473   689   689 E EGL_emulation: tid 689: eglCreateSyncKHR(1215): error 0x3004 (EGL_BAD_ATTRIBUTE)
09-24 11:19:18.581   676   701 W art     : Long monitor contention event with owner method=android.graphics.Bitmap com.android.server.wm.WindowManagerService.screenshotApplicationsInner(android.os.IBinder, int, int, int, boolean) from WindowManagerService.java:6223 waiters=0 for 171ms
09-24 11:19:18.631  1961  1980 E Surface : getSlotFromBufferLocked: unknown buffer: 0xac1c5ee0
09-24 11:19:18.684   794  1165 W EGL_emulation: eglSurfaceAttrib not implemented
09-24 11:19:18.684   794  1165 W OpenGLRenderer: Failed to set EGL_SWAP_BEHAVIOR on surface 0xde2eba80, error=EGL_SUCCESS
09-24 11:19:19.907   676  1391 I ActivityManager: START u0 {act=android.intent.action.MAIN cat=[android.intent.category.HOME] flg=0x10200000 cmp=com.android.launcher3/.Launcher} from uid 10016 on display 0
09-24 11:19:19.928  1961  1961 D StatusConnector: StatusConnector not bound!
09-24 11:19:19.928  1961  1961 D AndroidRuntime: Shutting down VM
09-24 11:19:19.932  1961  1961 E AndroidRuntime: FATAL EXCEPTION: main
09-24 11:19:19.932  1961  1961 E AndroidRuntime: Process: com.statusim, PID: 1961
09-24 11:19:19.932  1961  1961 E AndroidRuntime: java.lang.RuntimeException: Unable to destroy activity {com.statusim/com.statusim.MainActivity}: java.lang.IllegalArgumentException: Receiver not registered: com.centaurwarchief.smslistener.SmsReceiver@c37298f
09-24 11:19:19.932  1961  1961 E AndroidRuntime:       	at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:3831)
09-24 11:19:19.932  1961  1961 E AndroidRuntime:       	at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:3849)
09-24 11:19:19.932  1961  1961 E AndroidRuntime:       	at android.app.ActivityThread.-wrap5(ActivityThread.java)
09-24 11:19:19.932  1961  1961 E AndroidRuntime:       	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1398)
09-24 11:19:19.932  1961  1961 E AndroidRuntime:       	at android.os.Handler.dispatchMessage(Handler.java:102)
09-24 11:19:19.932  1961  1961 E AndroidRuntime:       	at android.os.Looper.loop(Looper.java:148)
09-24 11:19:19.932  1961  1961 E AndroidRuntime:       	at android.app.ActivityThread.main(ActivityThread.java:5417)
09-24 11:19:19.932  1961  1961 E AndroidRuntime:       	at java.lang.reflect.Method.invoke(Native Method)
09-24 11:19:19.932  1961  1961 E AndroidRuntime:       	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
09-24 11:19:19.932  1961  1961 E AndroidRuntime:       	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
09-24 11:19:19.932  1961  1961 E AndroidRuntime: Caused by: java.lang.IllegalArgumentException: Receiver not registered: com.centaurwarchief.smslistener.SmsReceiver@c37298f
09-24 11:19:19.932  1961  1961 E AndroidRuntime:       	at android.app.LoadedApk.forgetReceiverDispatcher(LoadedApk.java:780)
09-24 11:19:19.932  1961  1961 E AndroidRuntime:       	at android.app.ContextImpl.unregisterReceiver(ContextImpl.java:1195)
09-24 11:19:19.932  1961  1961 E AndroidRuntime:       	at android.content.ContextWrapper.unregisterReceiver(ContextWrapper.java:576)
09-24 11:19:19.932  1961  1961 E AndroidRuntime:       	at com.centaurwarchief.smslistener.SmsListenerModule.unregisterReceiver(SmsListenerModule.java:44)
09-24 11:19:19.932  1961  1961 E AndroidRuntime:       	at com.centaurwarchief.smslistener.SmsListenerModule.onHostDestroy(SmsListenerModule.java:60)
09-24 11:19:19.932  1961  1961 E AndroidRuntime:       	at com.facebook.react.bridge.ReactContext.onHostDestroy(ReactContext.java:189)
09-24 11:19:19.932  1961  1961 E AndroidRuntime:       	at com.facebook.react.XReactInstanceManagerImpl.moveToBeforeCreateLifecycleState(XReactInstanceManagerImpl.java:603)
09-24 11:19:19.932  1961  1961 E AndroidRuntime:       	at com.facebook.react.XReactInstanceManagerImpl.onHostDestroy(XReactInstanceManagerImpl.java:535)
09-24 11:19:19.932  1961  1961 E AndroidRuntime:       	at com.facebook.react.XReactInstanceManagerImpl.onHostDestroy(XReactInstanceManagerImpl.java:542)
09-24 11:19:19.932  1961  1961 E AndroidRuntime:       	at com.facebook.react.ReactActivityDelegate.onDestroy(ReactActivityDelegate.java:128)
09-24 11:19:19.932  1961  1961 E AndroidRuntime:       	at com.facebook.react.ReactActivity.onDestroy(ReactActivity.java:72)
09-24 11:19:19.932  1961  1961 E AndroidRuntime:       	at com.statusim.MainActivity.onDestroy(MainActivity.java:62)
09-24 11:19:19.932  1961  1961 E AndroidRuntime:       	at android.app.Activity.performDestroy(Activity.java:6407)
09-24 11:19:19.932  1961  1961 E AndroidRuntime:       	at android.app.Instrumentation.callActivityOnDestroy(Instrumentation.java:1142)
09-24 11:19:19.932  1961  1961 E AndroidRuntime:       	at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:3818)
09-24 11:19:19.932  1961  1961 E AndroidRuntime:       	... 9 more
09-24 11:19:19.977   794  1165 E Surface : getSlotFromBufferLocked: unknown buffer: 0xe82ecdb0
09-24 11:19:20.044   676  1950 I OpenGLRenderer: Initialized EGL, version 1.4
09-24 11:19:20.085  1036  1220 W EGL_emulation: eglSurfaceAttrib not implemented
09-24 11:19:20.085  1036  1220 W OpenGLRenderer: Failed to set EGL_SWAP_BEHAVIOR on surface 0xe0f20b60, error=EGL_SUCCESS
09-24 11:19:20.090   676  1950 W EGL_emulation: eglSurfaceAttrib not implemented
09-24 11:19:20.090   676  1950 W OpenGLRenderer: Failed to set EGL_SWAP_BEHAVIOR on surface 0xda11bae0, error=EGL_SUCCESS
09-24 11:19:20.632  1036  1220 W OpenGLRenderer: Incorrectly called buildLayer on View: ShortcutAndWidgetContainer, destroying layer...
09-24 11:19:20.632  1036  1220 W OpenGLRenderer: Incorrectly called buildLayer on View: ShortcutAndWidgetContainer, destroying layer...
09-24 11:19:29.934   676   695 W ActivityManager: Activity destroy timeout for ActivityRecord{ae73e28 u0 com.statusim/.MainActivity t272 f}
09-24 11:19:38.062    99    99 D Genyd   : Received Set Clipboard
```

--- a/package.json
+++ b/package.json
@@ -29,7 +29,7 @@
     "react": "^15.3.1",
     "react-native": "^0.33.0",
     "react-native-action-button": "^1.1.11",
-    "react-native-android-sms-listener": "github:alwx/react-native-android-sms-listener#listener-bugfix",
+    "react-native-android-sms-listener": "github:adrian-tiberius/react-native-android-sms-listener#listener-bugfix",
     "react-native-camera": "git+https://github.com/lwansbrough/react-native-camera.git",
     "react-native-circle-checkbox": "github:paramoshkinandrew/ReactNativeCircleCheckbox",
     "react-native-contacts": "^0.2.4",

